feat(config): Add FCrDNS checker

Axelen123 opened the pull request:
Closes #431. This PR implements dynamic verification of bot IPs using DNS records. For details regarding how it works, see the documentation I have added. I am not sure if the way I added a new `algorithm` is the best way to implement this. Let me know if there is a better way.

- [x] Added a description of the changes to the `[Unreleased]` section of docs/docs/CHANGELOG.md
- [x] Added test cases to [the relevant parts of the codebase](https://anubis.techaro.lol/docs/developer/code-quality)
- [ ] Ran integration tests `npm run test:integration` (unsupported on Windows, please use WSL)

Axelen123 committed:
implement fcrdns challenge

Axelen123 committed:
Merge remote-tracking branch 'origin/main'

Axelen123 committed:
use algorithm instead

Axelen123 committed:
add documentation

Axelen123 committed:
Merge remote-tracking branch 'origin/main' into fcrdns

Axelen123 committed:
implement as a checker instead

Axelen123 committed:
Merge remote-tracking branch 'origin/main' into fcrdns

Axelen123 committed:
cleanup

Axelen123 commented on lib/policy/checker/checker.go:
```diff
@@ -19,15 +19,12 @@ type List []Impl
 			return ok, err
 		}
-		if ok {
-			return ok, nil
-		}
 	}
 
-	return false, nil
+	return true, nil
```
This change was made to ensure that we don't try to do any unnecessary reverse DNS checks if other checkers (such as user agent or CEL) fail. I assumed this was already the case and I am not sure if changing this will have unintended consequences.

Axelen123 committed:
Merge remote-tracking branch 'origin/main' into fcrdns

Axelen123 committed:
update documentation

Xe reviewed (approved):
Approved modulo the change to checker.List#Check

Axelen123 committed:
add fcrdns function to cel

Axelen123 committed:
revert multiple checker behavior

Axelen123 committed:
fully revert multi checker behavior

Axelen123 committed:
Merge remote-tracking branch 'origin/main' into fcrdns

Axelen123 commented:
I have added CEL bindings and reverted to the previous checker behavior.

Axelen123 committed:
fix formatting

Xe committed:
chore(lib/policy): run go tool goimports

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe committed:
docs(CHANGELOG): update changelog notes

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe committed:
feat(data/crawlers): add weight to clients that lie

If a client claims to be Googlebot but isn't from Google, that's kinda
suspicious and should be treated as such.

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe committed:
Merge branch 'main' into fcrdns

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe commented:
Thanks much! This is gonna let us do a lot of fun things :)

Xe committed:
chore: spelling

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe committed:
chore(data/crawlers/googlebot): fix typo

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe committed:
chore: fix spelling errors

Signed-off-by: Xe Iaso <me@xeiaso.net>

Xe committed:
tests(lib): inject fcrdns into the test contexts

Signed-off-by: Xe Iaso <me@xeiaso.net>

Axelen123 commented:
I was reminded of this PR recently. Did you forget to merge it or is there something else?
Let me know if there is anything else I need to do (and thank you for the other commits)

Xe commented:
I forgot about this, I'm going to rebase after work.
